### DISCUSSION TITLE
How to get un-normalized regression predictions

### DISCUSSION THREAD

**Discussion by Aceticia** [manujosephv/pytorch_tabular#99] [Q&A] [answered]
I've trying to do regressions following the code from the tutorial, but the predicted values for my dataset seem to all very close to 0 (the actual values are in the range 50-3000). I suspect that this is not converted back from normalized values. Did I miss any important steps?

**Comment by manujosephv** [comment, marked as answer]
PyTorch Tabular does not use any target transformation by default. There is a way to use a target_transformer in the `fit` method, but if you do that, they are reverse transformed before reporting the predictions.

What might be happening in your case can be that the model might be struggling to capture the scale of the data. There are two things you can try:
1. Use `target_range` parameter in `ModelConfig` and specify the range you expect your output to be in. This uses a sigmoid layer to force the output to be in between the given range. But this is only a good way if you expect something similar to a normal distribution for your target.
2. Scale the target. Even something as simple as dividing by the mean of the target should help the model to learn better.

**Reply by miaozi-hao** [reply]
Typically, after the model configuration is set up, the model becomes quite large. Then, the training usually stops after 3-5 epochs, and the predicted results are also around 0. Could this be due to the small size of the dataset?

**Reply by manujosephv** [reply]
I don't know the details of your problem, but it does look like the model hasn't learned anything, or the training was stopped prematurely. May be try with lower learning rates, or disable early stopping and run for set number of epochs etc. 

And small datasets are typically harder to learn from, How small are we talking about?

**Reply by miaozi-hao** [reply]
sure for example 200-300
